tests/datadotworld/client/test_client.py

```diff
@@ -190,3 +237,43 @@ def test_download_datapackage_existing_dest_dir(
             calling(api_client.download_datapackage).with_args(
                 dataset_key, config.cache_dir),
             raises(ValueError))
+
+    def test_download_dataset(self, api_client, dataset_key, download_api):
+        api_client.download_dataset(dataset_key)
+        assert_that(download_api.download_dataset,
+                    called().times(1).with_args('agentid', 'datasetid'))
+
+    def test_download_file(self, api_client, dataset_key, download_api):
+        api_client.download_file(dataset_key, 'file')
+        assert_that(download_api.download_file,
+                    called().times(1).with_args('agentid', 'datasetid', 'file'))
+
+    def test_sql(self, api_client, dataset_key, sql_api):
+        api_client.sql(dataset_key, 'query')
+        assert_that(sql_api.sql_post,
```

Comment:
@andela-gokerayi The reason you have a busted test is that `sql_api` here is not being used by `api_client.sql`. Instead `api_client.sql` is creating an instance of `ContentNegotiatingApiClient` on the fly. So, when the tests run, instead of invoking `sql_api.sql_post`, they are invoking `sql_post` on a concrete instance of `ContentNegotiatingApiClient` which results in a failed network request.

One way to fix it would be to inject your mock via `**kwargs` when you call `api_client.sql`.

Reply:
Fixed!